doc/ipv6/ipv6_link_local.md

```diff
@@ -0,0 +1,378 @@
+
+Changes done in NeighOrch:
+
+* Add the interface parameter to the next hop object key. This allows to add multiple IPv6 link local next hops with same IPv6 address. This change was originally implemented as part of the VRF implementation.
+
+```
+struct NextHopKey
+{
+    IpAddress           ip_address;     // neighbor IP address
+    string              alias;          // incoming interface alias
+}
+```
+
+```
+sonic# show ipv6 route
+
+S>*  2222::/64 [1/0] via fe80::5054:ff:fe03:6175, Ethernet0, 00:00:04
+  *                  via fe80::5054:ff:fe03:6175, Ethernet4, 00:00:04
+  *                  via fe80::5054:ff:fe03:6175, Ethernet8, 00:00:04
+
+127.0.0.1:6379> keys *NEIGH*
+ 1) "NEIGH_TABLE:Ethernet8:fe80::5054:ff:fe03:6175"
+ 2) "NEIGH_TABLE:Ethernet0:fe80::5054:ff:fe03:6175"
+ 3) "NEIGH_TABLE:Ethernet4:fe80::5054:ff:fe03:6175"
+
+```
+
+### 3.3.2 NeighSyncd
+
+Neighsyncd is updated to learn the IPv6 link-local neighbors. All link-local neighbors that are dynamically learned are programmed to ASIC_DB.
+
+### 3.3.3 IntfMgr
+In the absence of manually configured addresses on an interface, to enable/disable IPv6 mode, IntfMgr is modified to handle changes to the CONFIG DB `'ipv6_use_link_local_only'` setting. The IPv6 mode is enforced using the Linux system control utility to set the `'disable_ipv6'` parameter. The default value is also set so that new interfaces are created with default disable IPv6 setting.
+
+`sysctl -w net.ipv6.conf.default.disable_ipv6=1`
+
+Since the Linux kernel auto-generates the IPv6 link-local address per interface, netlink events for IPv6 address addition and deletion are handled by the IntfMgr. All netlink messages other than RTM_NEWADDR, RTM_DELADDR are ignored. It also ignores all addresses other than IPv6 link-local addresses.
+
+- On receiving IPv6 Link local address add event on an interface, entry is created in both APP DB and STATE DB. Orchagent then creates the L3 RIF based on these entries. This is needed to ensure that the L3 routing can happen even with out any global IPv6 address configured on the interface.
+- On receiving IPv6 Link local address delete event on interface, the corresponding entries are deleted from APP DB and STATE DB. Orchagent then deletes the L3 RIF from the ASIC DB.
+
+## 3.4 SyncD
+
+## 3.5 SAI
+
+## 3.6 CLI
+### 3.6.1 Data Models
+
+N/A
+
+### 3.6.2 Configuration Commands
+
+```
+# config interface ipv6 enable use-link-local-only
+Usage: config interface ipv6 [OPTIONS] COMMAND [ARGS] 
+
+  Enable or Disable IPv6 processing on interface
+
+Options:
+  --help  Show this message and exit.
+
+Commands:
+  disable  Disable IPv6 processing on interface
+  enable   Enable IPv6 processing on interface
+
+```
+
+# 4 Flow Diagrams

```

Comment:
Please add this section with the flows, especially on LL IP2ME  and /10 route installation and deletion

Reply:
Done